recipe/meta.yaml

```
@@ -1,45 +1,56 @@
 {% set name = "cuda-python" %}
-{% set version = "11.8.1" %}
+{% set version = "12.0.0" %}
 {% set major_version = version.split(".")[0] %}
 {% set minor_version = version.split(".")[1] %}
 {% set patch_version = version.split(".")[2] %}
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/NVIDIA/cuda-python/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: ef54ada5d607387f59950cfb3c3852e5816b62824402d289227b8066e8353489
+  sha256: 7df0f84c4b6210112353374a7d992bfd5bba9c97358b39ea03b61bd634ce9c8a
 
 build:
-  number: 2
-  skip: true  # [py<37 or osx or cuda_compiler_version != "11.8"]
+  number: 0
+  skip: true  # [py<37 or osx or cuda_compiler_version != "12.0"]
   script:
     - {{ PYTHON }} -m pip install . --no-deps -vv
+  script_env:
+    # Ensure that CUDA includes can be found by the host compiler
+    - export CUDA_HOME="${BUILD_PREFIX}/targets/{{ target_name }}"
   run_exports:
-    - cuda-python >=11.7.1,<12  # we need this because CUDA Python 11.7.1 has ABI breaks, see conda-forge/cuda-python-feedstock#15
-  ignore_run_exports:
-    - cudatoolkit
+    - cuda-python >=12.0.0
+  # TODO: Do we need to ignore anything in CUDA 12?
+  #ignore_run_exports:
+  #  - cuda-cudart
+  #  - cuda-nvrtc
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    - cuda-nvrtc-dev
+    - cuda-profiler-api
+    - sysroot_{{ target_platform }} 2.17  # [linux]
   host:
```

Would it make sense to add `cuda-nvrtc-dev` & `cuda-profiler-api` to `build` & `host`?

This would pick up the shared libraries from `cuda-nvrtc-dev` during linkage checks / fixes by conda-build. Also would allow [`run_exports` to add `cuda-nvrtc` to `run`]( https://github.com/conda-forge/cuda-nvrtc-feedstock/blob/baea183c65beac3e32e68fa2499c5fcc6a736a63/recipe/meta.yaml#L67-L69 ). So we wouldn't need to do that ourselves

Plus this would simplify the change when dropping the packages from `build` in the future

Also do we need `cuda-cudart-dev` (since we have `cuda-cudart` in `run`)?

```suggestion
  host:
    - cuda-cudart-dev
    - cuda-nvrtc-dev
    - cuda-profiler-api
```

---

Great idea.